include/glow/Runtime/RuntimeTypes.h

```diff
@@ -103,12 +103,18 @@ using DAGListTy = std::vector<DAG>;
 /// member variable to it's correct BackendKind.
 class DeviceConfig {
   const BackendKind backendKind_;
+  std::string name_;
 
-protected:
+public:
   DeviceConfig(BackendKind kind) : backendKind_(kind) {}
+  DeviceConfig(BackendKind kind, std::string name)
+      : backendKind_(kind), name_(name) {}
 
-public:
   BackendKind getBackendKind() { return backendKind_; }
+
+  llvm::StringRef getName() { return name_; }
+  bool hasName() { return name_ != ""; }
+  void setName(llvm::StringRef name) { name_ = name; }
```

Comment:
nit: maybe another function to compare the names?

Reply:
@beicy what do you suggest?

Reply:
I thought we may need to compare if two DeviceConfig have the same backend name. But on second thought, we don't really need to do it, right?

Reply:
I don't think so, the BackendName is really just for logging anyway.